Packs/PAN-OS/Playbooks/PAN-OS_-_Configure_DNS_Sinkhole.yml

```diff
@@ -0,0 +1,1963 @@
+      "76_69_Yes": 0.61,
+      "76_75_No": 0.56,
+      "77_75_#default#": 0.11,
+      "82_59_#default#": 0.52,
+      "82_85_yes": 0.11
+    },
+    "paper": {
+      "dimensions": {
+        "height": 6045,
+        "width": 3590,
+        "x": -1440,
+        "y": -60
+      }
+    }
+  }
+inputs:
+- key: DNSSignatureSource
+  value:
+    simple: default-paloalto-dns
+  required: true
+  description: |-
+    The name of the signature source that contains the domains to sinkhole. It is used by the security profile to check if DNS requests for these domains are being sent through the firewall. Supported signature sources are: Custom EDLs of type Domain, Palo Alto Networks Content-delivered malicious domains, and DNS Security Categories.
+
+    To find your available DNS signature sources in PAN-OS, navigate to Objects > Security Profiles > Anti-Spyware.
+    Click on the name of an existing profile or click Add, then open the DNS Policies tab.
+
+    Note: using External Dynamic Lists (EDLs) or DNS Security as signature sources requires a "DNS Security" subscription. If you don't have one, you can keep the default value of "default-paloalto-dns", which is part of the Palo Alto Networks Content.
+  playbookInputQuery:
+- key: SecurityProfileName
+  value:
+    simple: Sinkhole Anti Spyware Profile
+  required: false
+  description: |-
+    The name of the Anti-Spyware Security Profile that will be applied to the DNS related rule, in order to filter the malicious domains and facilitate the sinkhole action. The profile has to be an anti-spyware profile that the user wants the playbook to modify. The playbook will apply this profile to a security rule currently allowing DNS traffic.
+    If a security profile with the specified name does not exist, it will be created automatically.
+    Afterwards, the playbook will modify the profile to sinkhole any traffic to the domains found in the DNS signature source defined in the playbook input.
+
+    If you're not sure, leave the default value of this input.
+  playbookInputQuery:
+- key: PrimaryInternalDNSServerIP
+  value: {}
+  required: true
+  description: The IP of the primary internal DNS server. Used to find a policy that matches the malicious DNS queries that go out from the internal DNS server to the firewall. The idea is to find a policy that allows DNS traffic in order to sinkhole that traffic when used for resolving malicious domains found in our signature source.
+  playbookInputQuery:
+- key: SecondaryInternalDNSServerIP
+  value: {}
+  required: false
+  description: Optional - the IP of the secondary internal DNS server. Used to find a policy that matches the malicious DNS queries that go out from the internal DNS server to the firewall. The idea is to find a policy that allows DNS traffic in order to sinkhole that traffic when used for resolving malicious domains found in our signature source.
+  playbookInputQuery:
+- key: PublicDNSServerIP
+  value:
+    simple: 8.8.8.8
```

Comment:
Do we want to keep Google's dns ip address?

Reply:
I think it's reasonable as a default value. Objection?